metacat-main/src/main/java/com/netflix/metacat/main/api/ApiFilter.java

```diff
@@ -65,14 +65,21 @@ public void doFilter(
         final String clientHost = httpServletRequest.getHeader("X-Forwarded-For");
         final String jobId = httpServletRequest.getHeader(MetacatRequestContext.HEADER_KEY_JOB_ID);
         final String dataTypeContext = httpServletRequest.getHeader(MetacatRequestContext.HEADER_KEY_DATA_TYPE_CONTEXT);
+        String apiVersion = null;
+        for (Object version: MetacatRequestContext.REST_API_PATH_PREFIX) {
+            if (httpServletRequest.getRequestURI().startsWith((String) version)) {
+                apiVersion = (String) version;
```

Comment:
I think this is pretty inefficient to do this on every request when the request is already being filtered and checked by the web framework. I think a possible cleaner design would be to instantiate multiple instances of this filter and bind them to the different paths and set the `apiVersion` in the constructor and use it here as a constant. That way once the filter is invoked you're already in the right path. Simply put you could have two filters one instantiated for `0` listening on `/mds/v0/*` and one instantiated for `1` listening on `/mds/v1/*` each would only be invoked when those paths are requested. No need to do extra checking on the path. That would be my suggestion or something similar that doesn't involve so much coding of constants and instead relies on configuration and re-use.

Reply:
As a total aside are we tied to this whole `/mds` thing? when we deprecate `v0` can we move it to something like `/api`? Might just be my OCD talking though haha